src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetDeserializer.cs

```diff
@@ -124,10 +126,133 @@ public virtual IEnumerable ReadResourceSet(ODataResourceSetWrapper resourceSet,
                     Error.Format(SRResources.TypeCannotBeDeserialized, elementType.FullName()));
             }
 
-            foreach (ODataResourceWrapper resourceWrapper in resourceSet.Resources)
+            Collection<object> coll = new Collection<object>();
+
+            if (resourceSet.ResourceSetType == ResourceSetType.ResourceSet)
+            {
+                foreach (ODataResourceWrapper resourceWrapper in resourceSet.Resources)
+                {
+                    coll.Add(deserializer.ReadInline(resourceWrapper, elementType, readContext));
+                }
+            }
+            else
+            {
+                foreach (ODataItemBase odataItemBase in resourceSet.Resources)
+                {
+                    ODataResourceWrapper resourceWrapper = odataItemBase as ODataResourceWrapper;
+
+                   if(resourceWrapper != null)
+                    {
+                        ODataDeletedResource deletedResource = resourceWrapper.ResourceBase as ODataDeletedResource;
+
+                        if (deletedResource != null)
+                        {
+                            coll.Add(DeSerializeDeletedEntity(readContext, deletedResource));
+                        }
+                        else
+                        {
+                            coll.Add(deserializer.ReadInline(resourceWrapper, elementType, readContext));
+                        }
+                    }
+                    else
+                    {
+                        ODataDeltaLinkWrapper deltaLinkWrapper = odataItemBase as ODataDeltaLinkWrapper;
+
+                        IEdmDeltaLinkBase deltaLink = DeserializeDeltaLink(readContext, deltaLinkWrapper);
+                        coll.Add(deltaLink);
+                    }
+                    
+                }
+            }
+
+            return coll;
+        }
+
+
+        private static IEdmDeltaLinkBase DeserializeDeltaLink(ODataDeserializerContext readContext, ODataDeltaLinkWrapper deltaLinkWrapper)
+        {
+            ODataDeltaLinkBase deltalink = deltaLinkWrapper.DeltaLink;
+
+            if (deltalink == null)
+            {
+                throw new ODataException("Deleted link not present");
+            }
+
+            IEdmModel model = readContext.Model;
+
+            if (model == null)
+            {
+                throw Error.Argument("readContext", SRResources.ModelMissingFromReadContext);
+            }
+
+            IEdmStructuredType actualType = model.FindType(deltalink.TypeAnnotation.TypeName) as IEdmStructuredType;
```

Comment:
>deltalink.TypeAnnotation.TypeName [](start = 59, length = 33)

do delta links generally have a type annotation?  I wouldn't think so.  You may have to default to using the type of the resourcesetwrapper as the expected type for the (common) case of the deltaLink not having a type annotation.